src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/service/TaskService.kt

```diff
@@ -156,7 +156,7 @@ class TaskService(
     }
 
     val assigneeUser = when (val assigneeUserResult = userService.updateUserFromCommunityApiById(userToAllocateToId, ServiceName.approvedPremises)) {
-      is AuthorisableActionResult.Success -> assigneeUserResult.entity
+      is AuthorisableActionResult.Success -> assigneeUserResult.entity.user!!
```

Comment:
i know this will never happen currently because the updateUserFromCommunityApiById will throw an exception, but i think it's probably best for consistency with how we handle GetUserResponse to make line 159 something like the following (psuedo code, i'm not sure if you can do things like this in a single when line?)

```is AuthorisationActionResult.Success && assigneeUserResult.staffRecordFound -> assigneeUserResult.entity.user!!```

Reply:
Yep, that makes sense to me. I guess it is a bit odd how we return 'Success' despite staff not found but that's a pattern we're committed to now for this profile stuff :)